patterns/ink/InkMetadataRune.ts

```diff
@@ -35,14 +35,45 @@ export class InkMetadataRune<out U> extends Rune<InkMetadata, U> {
 
   codecs = this
     .into(ValueRune)
-    .access("V3", "types")
+    .access("types")
     .map(transformTys)
     .access(0)
 
+  $event = Rune
+    .tuple([
+      this.into(ValueRune).access("spec", "events"),
+      this.codecs,
+    ])
+    .map(([events, codecs]) =>
+      $.taggedUnion(
+        "type",
+        events.map((event) =>
+          $.variant(
+            event.label,
+            $.object(...event.args.map((arg) => $.field(arg.label, codecs[arg.type.type]!))),
+          )
+        ),
+      )
+    )
+    .into(CodecRune)
+
+  $error<C extends Chain, U, X>(
```

Comment:
On second thought... is this any different from [`unhandleFailed`](https://github.com/paritytech/capi/blob/794edd47ee685b6046676eea88173606793f12d2/fluent/ExtrinsicsEventsRune.ts#L14)?

Reply:
Might have implemented this before we had `unhandleFailed`. If they are in fact the same, then we can likely scrap this, and `into` the `ExtrinsicEventsRune`.

Reply:
Might have implemented this before we had `unhandleFailed`. If they are in fact the same, then we can likely scrap this, and `into` the `ExtrinsicEventsRune`.

Reply:
I'm not very familiar with that API.
I'll investigate